### DISCUSSION TITLE
Is it possible to use animation or integrate unistyles with reanimated?

### DISCUSSION THREAD

**Discussion by NikPyzhevsky** [jpudysz/react-native-unistyles#286] [Feature Requests]
Is there any way we can do this using plugins or anything else?

**Comment by jpudysz** [comment]
You can mix Reanimated with Unistyles freely. What do you mean? What is your use case?

**Reply by NikPyzhevsky** [reply]
I mean something like that : 
```
const stylesheet = createStyleSheet((theme) => ({
  root: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.shared.background.primary,
  },
  animated_style: (progress: SharedValue<number>) => ({
    opacity: progress.value
  }), 
}));
```
Although now looking at core reanimated it looks like it would be easier to just try to make a babel plugin somehow or gut core reanimated but they ask not to do this

**Reply by jpudysz** [reply]
I'm also against it, what are the benefits? I would need to maintain another integration which is not perfect.

**Reply by NikPyzhevsky** [reply]
I am just playing around to find a way to use animations without hooks. Mb there is some way to integrate animations with stylesheet(not necessarily reanimated).

**Reply by jpudysz** [reply]
ATM no, but with Unistyles 3.0 it will be possible (different architecture). As of now I don't have any plans to implement it
